platform/kube/conversion.go

```diff
@@ -83,11 +83,16 @@ func convertPort(port v1.ServicePort) *model.Port {
 
 func convertService(svc v1.Service) *model.Service {
 	ports := make([]*model.Port, 0)
-	addr := ""
+	addr, external := "", false
 	if svc.Spec.ClusterIP != "" && svc.Spec.ClusterIP != v1.ClusterIPNone {
 		addr = svc.Spec.ClusterIP
 	}
 
+	if svc.Spec.ExternalName != "" {
```

Comment:
You should compare `svc.Spec` service type, like in https://github.com/istio/manager/blob/master/platform/kube/controller.go#L520. There is some dead code there that needs to be updated as well.